src/core/rendering.js

```diff
@@ -290,13 +290,17 @@ p5.prototype.noCanvas = function() {
  * @param  {HTMLCanvasElement} [canvas]
  * @return {p5.Graphics} offscreen graphics buffer
  */
-p5.prototype.createGraphics = function(w, h, renderer, canvas) {
-  if (arguments[2] instanceof HTMLCanvasElement) {
-    renderer = constants.P2D;
-    canvas = arguments[2];
+p5.prototype.createGraphics = function(w, h, ...args) {
+/**
+  * args[0] is expected to be renderer
+  * args[1] is expected to be canvas
+  */
+  if (args[0] instanceof HTMLCanvasElement) {
+    args[0] = constants.P2D;
+    args[1] = args[0];
```

Comment:
I believe this is a bug: `args[1]` will also be `constants.P2D` because you are setting it equal to `args[0]` after resetting that to be `constants.P2D`.

Reply:
Absolutely! I was thinking the same. Maybe using a temporary variable could do the trick? What do you reckon?

Reply:
That would work, or alternatively just setting `args[1]` before `args[0]`.